pkg/transforms/db/interfaces/store.go

```diff
@@ -0,0 +1,42 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+
+// interfaces establishes the contract required for any implementation of the export store functionality in a database provider.
+package interfaces
+
+import "github.com/edgexfoundry/app-functions-sdk-go/internal/store/models"
+
+// StoreClient establishes the contracts required to persist exported data before being forwarded.
+type StoreClient interface {
+	// NewStoredObject establishes an object inside the persistent store.
+	NewStoredObject(object interface{}) (o models.StoredObject, err error)
```

Comment:
What are you expecting `object` to be here? I would expect new to have parameters for the required StoredObject fields, which are: AppServiceKey, Payload, PipelinePosition and Version. Others can be set optionally after call to New.

Reply:
I was trying to build to the spec defined in the call, but if we just want a constructor... do we need a constructor? If so:

1. Does that constructor need to be part of the database interface? I'd think it would make more sense outside of that.

2. If it has an exported constructor, should it be an unexported type?

If it doesn't need a constructor, let's just remove this function and use the `Store` method to create.

Reply:
I added a commit with the constructor idea in place. If you don't like that, I can revert back to the exported type with no constructor pretty easily.

Reply:
Yes, I think constructors are good to have, you are right the constructor should be separate from this interface. It should be implemented in this package.